witchcraft/refreshable/refreshable_file.go

```diff
@@ -35,73 +34,58 @@ type fileRefreshable struct {
 }
 
 // NewFileRefreshable returns a new Refreshable whose current value is the bytes of the file at the provided path.
+// The duration configures how often the refreshable check the file
 // Calling this function also starts a goroutine which updates the value of the refreshable whenever the specified file
 // is changed. The goroutine will terminate when the provided context is done or when the returned cancel function is
 // called.
-func NewFileRefreshable(ctx context.Context, filePath string) (r Refreshable, rErr error) {
+func NewFileRefreshable(ctx context.Context, filePath string, duration time.Duration) (Refreshable, error) {
 	initialBytes, err := ioutil.ReadFile(filePath)
 	if err != nil {
-		return nil, werror.Wrap(err, "failed to create file-based Refreshable because file could not be read", werror.SafeParam("filePath", filePath))
+		return nil, werror.WrapWithContextParams(ctx, err, "failed to create file-based Refreshable because file could not be read", werror.SafeParam("filePath", filePath))
 	}
 	fRefreshable := &fileRefreshable{
 		innerRefreshable: NewDefaultRefreshable(initialBytes),
 		filePath:         filePath,
 		fileChecksum:     sha256.Sum256(initialBytes),
 	}
-	if err := fRefreshable.watchForChanges(ctx); err != nil {
-		return nil, err
-	}
+	fRefreshable.watchForChangesAsync(ctx, duration)
 	return fRefreshable, nil
 }
 
-func (d *fileRefreshable) watchForChanges(ctx context.Context) error {
-	watcher, err := fsnotify.NewWatcher()
+func (d *fileRefreshable) watchForChangesAsync(ctxOuter context.Context, duration time.Duration) {
```

Comment:
I don't think you should need to use `ctxOuter`, scoping should do the right thing with `ctx`

Reply:
Just for my own readability. Done